api/v1alpha1/stargate_types.go

```diff
@@ -79,14 +79,17 @@ type StargateTemplate struct {
 
 	// AllowStargateOnDataNodes allows Stargate pods to be scheduled on a worker node already hosting data pods for this
 	// datacenter. The default is false, which means that Stargate pods will be scheduled on separate worker nodes.
+	// Note: if the datacenter pods have HostNetwork:true, then the Stargate pods will inherit of it, in which case it
+	// is possible that Stargate nodes won't be allowed to sit on data nodes even if this property is set to true,
+	// because of port conflicts on the same IP address.
```

Comment:
Can we work around this by using non-default ports?

Reply:
I can make the network settings for the CassandraDatacenter configurable in #94.

Reply:
I am not sure this would be a good idea. We have several port conflicts between a Cassandra pod and a Stargate pod; here are the ports required by both pod templates:

```
cassandra: 7000 7001 7199 8080                               9042 9103 9142 9160
stargate:  7000 7001      8080 8081 8082 8084 8085 8090 8609 9042
```

So the ports creating conflicts when HostNetwork=true are: 7000, 7001, 8080, 9042.

I guess 8080 is used for probes? The other 3 are for internode communication (7000, 7001) and client communication (9042). I don't know if these can be changed on a per-node basis; last time I checked, this is only possible in Cassandra 4.0, see [CASSANDRA-7544](https://issues.apache.org/jira/browse/CASSANDRA-7544).